docs/guide/config_dynamic.md

```diff
@@ -44,6 +44,16 @@ $myParamSettingAdmin = $configResolver->getParameter( 'my_param', 'myapp', 'admi
 // $myParamSetting's value will be 'another value'
 ```
 
+In debug mode, ConfigResolver detects if parameters were loaded prior to initialization of SiteAccess to warn about issues that are hard to resolve.
+It will log all instances of `getParameter()` that may be used unsafely.
+
+Additionally, for class-based services, the warning appears even if a service is recognized by `DynamicSettingsListener` as `$updatableServices` (which is safe).
+
+To avoid issues:
+- Do not use ConfigResolver in service factory.
```

Comment:
Hmm, this can be misleading.
Where did you take this from?

Reply:
I picked it from [this comment](https://github.com/ezsystems/ezpublish-kernel/pull/2636/files#diff-26112d3c76ce033446c6bf175dbdae01R323), but I may have misinterpreted something. Your suggestion would be to delete this line?

Reply:
So maybe then `Try to avoid eager usage of config resolver (e.g. in service factories)`